parser/parser/cc/lexer.rl

```diff
@@ -228,6 +229,33 @@ int lexer::compare_indent_level(token_t left, token_t right) {
     return 0;
 }
 
+size_t lexer::line_start(token_type type, size_t beginPos) {
+    if (type == token_type::tNL) {
+        // Doesn't make sense to ask for line start of newline character
+        return SIZE_MAX;
```

Comment:
`newline_s` was only ever `nullptr` before a `tNL` token had been read. that would only happen if we were on the first line, which means that the cur_line_start would be `0`

the new behavior is that `newline_s` is never consulted, instead we compute the list of tNL indices. if we were in the old case where `newline_s` was nullptr, now what will happen is that `beginPos` will be between `0` and whatever the first `tNL` is, and so return `0` (like before).

in the two lines you mention (601, 604) we always hard code the first argument (`token_type`) to `error` or `eof` respectively, so this `type == tNL` case will never get reached, and it will behave exactly like the old behavior (only consult `beginPos`, inheriting the correct handling for start-of-line behavior mentioned above.

lmk if this clears things up, but I'm pretty sure after re-reading this that the code behaves identically to before.